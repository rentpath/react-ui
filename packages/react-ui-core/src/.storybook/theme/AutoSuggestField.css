.AutoSuggestField {
  display: flex;
  margin: 75px;
}

.AutoSuggestField .Dropdown,
.AutoSuggestField .Field {
  margin: 0;
  width: 300px;
}

.AutoSuggestField .Button {
  height: 40px;
  position: relative;
  left: -70px;
  top: 5px;
}

.AutoSuggestField .Card {
  height: auto;
  width: auto;
  text-align: left;
}

.AutoSuggestField .List {
  padding: 0;
}

.AutoSuggestField .ListItem {
  padding: 5px 0;
}

.AutoSuggestField .Highlighter-Match {
  background: none;
}

.AutoSuggestField .Menu {
  padding-left: 5px;
}

.clear .Button {
  left: -40px;
}
