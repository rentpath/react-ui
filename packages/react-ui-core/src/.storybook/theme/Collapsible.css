.Collapsible_Items-visible {
  display: block;
}

.Collapsible_Items-hidden {
  display: none;
}

.Collapsible .Label {
  min-width: 200px;
}
