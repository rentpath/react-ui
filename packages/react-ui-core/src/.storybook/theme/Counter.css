.Counter {
  display: flex;
  white-space: nowrap;
  padding: 10px;
  align-items: center;
}

.Counter .Label {
  display: inline-block;
  white-space: nowrap;
  padding-right: 10px;
}

.Counter_Controls {
  display: inline-block;
  white-space: nowrap;
}

.Counter .Text {
  display: inline-block;
  text-align: center;
  padding: 0 20px;
}

.Counter_Button span {
  color: white;
  border-radius: 5px;
  padding: 5px 15px;
  display: inline-block;
  background: #0175D7;
  cursor: pointer;
}

.Counter_Button .Button {
  border-radius: 50%;
  width: 38px;
}

.Counter .Button {
  color: #ffffff;
  font-weight: bold;
}
