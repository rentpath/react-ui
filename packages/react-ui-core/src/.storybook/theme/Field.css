.Input,
.Select,
.Textarea {
  background-color: #ffffff;
  box-shadow: 0 0.5px 1px 0 rgba(0, 0, 0, 0.08);
  border: solid 1px #e9e9e9;
  border-radius: 4px;
  padding: 16px 10px;
  font-size: 13px;
}

.Field {
  margin: 16px 0;
  display: flex;
  max-width: 600px;
  color: #000;
  align-items: center;
}

.Field .Label {
  padding-right: 10px;
}
