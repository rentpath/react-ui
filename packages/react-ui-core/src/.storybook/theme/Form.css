.Form {
  min-width: 500px;
}

.Form .Field {
  color:#000;
}

.Form .Text {
  text-transform: uppercase;
  padding-bottom: 10px;
  margin-bottom: 10px;
}

.Form .Label {
  flex: 1;
}

.Form .Input,
.Form .Select,
.Form .Textarea {
  flex: 4;
}
