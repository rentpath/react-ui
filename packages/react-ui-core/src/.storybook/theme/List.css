.List {
  display: flex;
  padding: 20px;
}

.List-horizontal {
  flex-direction: row;
  justify-content: space-around;
}

.List-horizontal .ListItem {
  width: 100px;
}

.List-vertical {
  flex-direction: column;
}

.ListItem {
  cursor: pointer;
}

.ListItem-highlight {
  color: #AAA;
}
