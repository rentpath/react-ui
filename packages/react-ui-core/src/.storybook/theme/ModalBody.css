.ModalBody {
  min-width: 300px;
  padding: 20px;
  background: #FFF;
  flex-direction: column;
  border-radius: 5px;
  box-shadow: 1px 1px 5px #666;
  position: relative;
}
