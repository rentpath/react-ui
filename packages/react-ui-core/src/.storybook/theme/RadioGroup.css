.RadioGroup {
  border-radius: 5px;
  padding: .5rem 1rem;
}
