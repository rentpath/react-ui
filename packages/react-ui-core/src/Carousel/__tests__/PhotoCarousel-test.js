import React from 'react'
import { shallow, mount } from 'enzyme'
import LazyLoad from 'react-lazyload'
import renderer from 'react-test-renderer'
import theme from './mocks/theme'
import ThemedPhotoCarousel from '../PhotoCarousel'

const PhotoCarousel = ThemedPhotoCarousel.WrappedComponent
const items = [
  {
    path: 'imgr/2576db62ffa153ebef00317a5c68a368/',
    caption: 'test 1',
  },
  {
    path: 'imgr/d56984e959a3feb1235f85ee202a0fc6/',
    caption: 'test 2',
  },
  {
    path: 'imgr/fd972eb03a0463c484580349ad5177b7/',
    caption: 'test 3',
  },
]

const itemsWithProps = [
  {
    path: 'imgr/2576db62ffa153ebef00317a5c68a368/',
    caption: 'test 1',
    itemProps: {
      'data-tag_item': 'firstPhoto',
    },
  },
]

describe('PhotoCarousel', () => {
  it('applies a base theme to the root node', () => {
    const wrapper = shallow(
      <PhotoCarousel
        items={items}
        theme={theme}
        server="https://image.rent.com/"
        dimensions="400-200"
      />
    )
    const className = wrapper.prop('className')
    expect(className).toEqual('PhotoCarousel')
  })

  it('applies prop `className` to the root node', () => {
    const wrapper = shallow(
      <PhotoCarousel
        items={items}
        theme={theme}
        className={'Foo'}
        server="https://image.rent.com/"
        dimensions="400-200"
      />
    )
    const className = wrapper.prop('className')
    expect(className).toContain('Foo')
  })

  it('allows an object to be added to item if needed', () => {
    const wrapper = shallow(
      <PhotoCarousel
        items={itemsWithProps}
        server="https://image.rent.com/"
      />
    )
    const itemProp = wrapper.prop('items')
    expect(itemProp[0].itemProps['data-tag_item']).toBeTruthy()
  })

  describe('when no items', () => {
    it('renders an empty div when empty array', () => {
      const snap = renderer.create(<PhotoCarousel items={[]} server="https://image.rent.com/" />).toJSON()
      expect(snap).toMatchSnapshot()
    })

    it('renders an empty div when null', () => {
      const snap = renderer.create(<PhotoCarousel server="https://image.rent.com/" />).toJSON()
      expect(snap).toMatchSnapshot()
    })
  })

  it('renders correct url based on combination of props passed in', () => {
    const server = 'https://image.rent.com/'
    const dims = '400-200'
    const wrapper = mount(
      <PhotoCarousel
        items={items}
        theme={theme}
        server={server}
        dimensions={dims}
      />
    )
    const images = wrapper.find('img')
    expect(images.at(0).prop('src')).toEqual(`${server}${items[0].path}${dims}`)
    expect(images.at(1).prop('src')).toEqual(`${server}${items[1].path}${dims}`)
    expect(images.at(2).prop('src')).toEqual(`${server}${items[2].path}${dims}`)
  })

  it('lazyLoads images if required', () => {
    const wrapper = shallow(
      <PhotoCarousel
        theme={theme}
        items={items}
        server="https://image.rent.com/"
        dimensions="400-200"
        lazyLoad={{ once: true, height: 400, width: 300, offset: 200 }}
      />
    )
    expect(wrapper.find(<LazyLoad />)).toBeTruthy()
  })
})
