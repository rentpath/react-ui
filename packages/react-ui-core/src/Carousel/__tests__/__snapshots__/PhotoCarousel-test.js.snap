// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoCarousel when no items renders an empty div when empty array 1`] = `
<div
  className={undefined}
/>
`;

exports[`PhotoCarousel when no items renders an empty div when null 1`] = `
<div
  className={undefined}
/>
`;
